client/components/DataManagement/DataManagementRow/index.jsx

```diff
@@ -211,6 +211,39 @@ const DataManagementRow = ({
         );
     }, [testPlanVersions]);
 
+    const completedRequiredReports = testPlanVersion => {
```

Comment:
Better to not prematurely optimize but just wanted to add a note that I can imagine a scenario in which we have hundreds of DRAFT or CANDIDATE test plan versions and we would want to cache these results. These array operations on each test plan version on each rerender (since they are used as rendering conditionals) could grow expensive. One idea is that it could be a custom hook that takes `testPlanVersions` as an arg and returns a `hasCompletedRequiredReportsMap{testPlanVersionId -> bool}` using `useMemo` internally.
Again, just a note for the future.